frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -158,6 +168,15 @@ export const EmbedFrame = ({
     theme !== "transparent" && // https://github.com/metabase/metabase/pull/38766#discussion_r1491549200
     isParametersWidgetContainersSticky(visibleParameters.length);
 
+  // TODO: pass this as headerActions  from PublicDashboard ?
+  const saveAsPDF = async () => {
+    const cardNodeSelector = `#${DASHBOARD_PDF_EXPORT_ROOT_ID}`;
+    await saveDashboardPdf(cardNodeSelector, name!).then(() => {
```

Comment:
Done in 0733a05840b40038fb8c81690a2dbf8c1de53264, technically it should never happen, I tried to change the type of the props, but the issue is that we use EmbedFrame also when 404 or erroring, and in that case we don't have the dashboard name, I added a default file name just in case